.github/workflows/build_nightly.yml

```diff
@@ -0,0 +1,152 @@
+name: build_nightly
+
+on:
+  push:
+    branches:
+      - develop
+  workflow_dispatch:
+    inputs:
+      build_ios:
+        type: boolean
+        description: iOS
+        default: true
+      build_android:
+        type: boolean
+        description: Android
+        default: true
+      build_web:
+        type: boolean
+        description: Web
+        default: true
+
+defaults:
+  run:
+    working-directory: sample_app
+
+env:
+  FLUTTER_VERSION: "3.29.0"
+  FLUTTER_CHANNEL: stable
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_and_deploy_ios:
+    runs-on: macos-latest
+    timeout-minutes: 40
+    if: ${{ github.event_name == 'push' || inputs.build_ios == true }}
+    steps:
+      - name: Install Bot SSH Key
+        uses: webfactory/ssh-agent@v0.9.0
+        with:
+          ssh-private-key: ${{ secrets.BOT_SSH_PRIVATE_KEY }}
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          channel: ${{ env.FLUTTER_CHANNEL }}
+          flutter-version: ${{ env.FLUTTER_VERSION }}
+          cache: true
```

Comment:
We should also add the cache-key in order for the cache to work perfectly.
https://github.com/GetStream/stream-chat-flutter/blob/master/.github/workflows/stream_flutter_workflow.yml#L41